cli/src/commands/fix.rs

```diff
@@ -185,39 +191,49 @@ pub(crate) fn cmd_fix(
+        // --all-files flag or file arguments. The flag determines whether we diff the
+        // revision against its parents or just fix the entire tree. In either case, the
+        // file arguments filter that set of files.
         let tree = commit.tree()?;
-        let parent_tree = commit.parent_tree(tx.repo())?;
-        // TODO: handle copy tracking
-        let mut diff_stream = parent_tree.diff_stream(&tree, &matcher);
-        async {
-            while let Some(TreeDiffEntry {
-                path: repo_path,
-                values,
-            }) = diff_stream.next().await
-            {
-                let (_before, after) = values?;
-                // Deleted files have no file content to fix, and they have no terms in `after`,
-                // so we don't add any tool inputs for them. Conflicted files produce one tool
-                // input for each side of the conflict.
-                for term in after.into_iter().flatten() {
-                    // We currently only support fixing the content of normal files, so we skip
-                    // directories and symlinks, and we ignore the executable bit.
-                    if let TreeValue::File { id, executable: _ } = term {
-                        // TODO: Skip the file if its content is larger than some configured size,
-                        // preferably without actually reading it yet.
-                        let tool_input = ToolInput {
-                            file_id: id.clone(),
-                            repo_path: repo_path.clone(),
-                        };
-                        unique_tool_inputs.insert(tool_input.clone());
-                        paths.insert(repo_path.clone());
-                    }
+        let tree_values: Vec<(RepoPathBuf, Merge<Option<TreeValue>>)> = if args.all_files {
+            tree.entries_matching(&matcher)
+                .map(|(path, terms)| Ok::<_, BackendError>((path, terms?)))
+                .try_collect()?
+        } else {
+            // TODO: handle copy tracking by fixing all destination files, and only the
+            // source files that were otherwise modified. This will also need to extend to
+            // changed line computation once we support that.
+            // Deleted files have no file content to fix, and they have no terms in `after`,
+            // so we don't add any tool inputs for them. Conflicted files produce one tool
+            // input for each side of the conflict.
+            let parent_tree = commit.parent_tree(tx.repo())?;
+            async {
+                parent_tree
+                    .diff_stream(&tree, &matcher)
+                    .map(|entry| entry.values.map(|(_before, after)| (entry.path, after)))
+                    .try_collect()
+                    .await
+            }
+            .block_on()?
+        };
+        tree_values.into_iter().try_for_each(|(path, terms)| {
```

Comment:
nit: simply use `for (path, terms) in tree_values`?

Reply:
Thanks, that was probably left over from it being chained off the previous expression at some point.